### DISCUSSION TITLE
Data replay via server.js does not work

### DISCUSSION THREAD

**Discussion by Gordon-Shumway2** [JaapvanEkris/openrowingmonitor#12] [Q&A] [answered]
Hi, 

using v1beta_updates I tried to replay a short .csv file to calibrate data in order for my rower's ORM to produce similar numbers to EXR splits. _server.js_ was modifed to read the .csv:

> 

replayRowingSession(handleRotationImpulse, {
//  filename: 'recordings/2021/04/rx800_2021-04-21_1845_Rowing_30Minutes_Damper8.csv', // 30 minutes, damper 10
  filename: '/opt/openrowingmonitor/data/recordings/2023/01/2_31.csv',
  realtime: true,
  loop: true
})

The file seems to be read - with a incorrect filename an error is thrown in the logs. No such error with a correct filename.

Once ORM is restarted the log shows this: 

Jan 21 14:43:23 pirower npm[1933]: > openrowingmonitor@0.8.2 start /opt/openrowingmonitor
Jan 21 14:43:23 pirower npm[1933]: > node app/server.js
Jan 21 14:43:32 pirower npm[1946]: ==== Open Rowing Monitor 0.8.2 ====
Jan 21 14:43:32 pirower npm[1946]: bluetooth profile: FTMS Rower
Jan 21 14:43:32 pirower npm[1946]: Starting a just row session, no time or distance target set
Jan 21 14:43:33 pirower npm[1946]: webserver running on port 80
Jan 21 14:43:33 pirower npm[1946]: websocket client connected
Jan 21 14:43:33 pirower npm[1946]: *** WARNING: currentDt of NaN sec isn't between 0 and maximumStrokeTimeBeforePause (6 sec)
Jan 21 14:43:36 pirower npm[1946]: TS Linear Regressor, Division by zero prevented!
Jan 21 14:43:39 pirower npm[1946]: TS Linear Regressor, Division by zero prevented!
Jan 21 14:43:39 pirower npm[1960]: Gpio-service: pin number 17, polling interval 5 us, triggered on Up flank, minimal pulse time 50 us
Jan 21 14:44:01 pirower npm[1946]: *** WARNING: currentDt of 7 sec isn't between 0 and maximumStrokeTimeBeforePause (6 sec)
Jan 21 14:44:09 pirower npm[1946]: *** WARNING: currentDt of 8 sec isn't between 0 and maximumStrokeTimeBeforePause (6 sec)
Jan 21 14:44:18 pirower npm[1946]: *** WARNING: currentDt of 9 sec isn't between 0 and maximumStrokeTimeBeforePause (6 sec)
Jan 21 14:44:28 pirower npm[1946]: *** WARNING: currentDt of 10 sec isn't between 0 and maximumStrokeTimeBeforePause (6 sec)
Jan 21 14:44:39 pirower npm[1946]: *** WARNING: currentDt of 11 sec isn't between 0 and maximumStrokeTimeBeforePause (6 sec)
(...)

At this stage even after five minutes there are no updates on the GUI - standard "--" values are shown.
However once I pull on the handle values are displayed but those are values matching my pull, not the recorded data that I would expect.

Is recording/replay functionality broken on this branch? Or am I missing a config step? 

Thanks, 
Michael

**Comment by JaapvanEkris** [comment, marked as answer]
Have you looked at the contents of the csv file? It looks like it (based on the error messages) contains values from 1 to 11.....

**Comment by Gordon-Shumway2** [comment]
My bad - I fed ORM with a RowData .csv file rather than the raw data .csv file. Using the correct .csv file the playback functionality works as expected. 


Thank you and apologies for the extra work I caused, 

Michael

**Reply by JaapvanEkris** [reply]
No problem, always glad to help you!
